packages/core/src/tools/create-app-request-client.js

```diff
@@ -51,19 +51,13 @@ const createAppRequestClient = (input, options) => {
     httpBefores.push(disableSSLCertCheck);
   }
 
-  const httpOriginalAfters = [prepareResponse, logResponse];
+  const beforeAppAfterResponse = [prepareResponse, logResponse];
+  const afterAppAfterResponse = [throwForStatus];
 
-  if (app.authentication) {
-    if (
-      app.authentication.type === 'session' ||
-      (app.authentication.type === 'oauth2' &&
-        _.get(app, 'authentication.oauth2Config.autoRefresh'))
-    ) {
-      httpOriginalAfters.push(throwForStaleAuth);
-    }
-  }
```

Comment:
No more need for `throwForStaleAuth` as we now always call `throwForStatus` which also throws on 401 and we already reraise that as a `RefreshAuthError` on server/python-side for both cases.